src/aas-api-webapp-repository/Controllers/AasRepositoryApi.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.Annotations;
+using System.ComponentModel.DataAnnotations;
+using System.Threading.Tasks;
+using AAS.API.Registry.Attributes;
+
+using Microsoft.AspNetCore.Authorization;
+using AAS.API.Models;
+using Microsoft.Extensions.Configuration;
+using AAS.API.Repository;
+using AAS_Services_Support.ADT_Support;
+using AutoMapper;
+
+namespace AAS.API.Registry.Controllers
+{
+    [Authorize]
+    [ApiController]
+    [Route("api/v1/shells")]
+    public class AasRepositoryApi : Controller
+    {
+
+        private IConfiguration _configuration;
+
+        private AASRepository repository;
+        private IMapper _mapper;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public AasRepositoryApi(IConfiguration config, IAdtInteractions adtInteractions, IMapper mapper) : base()
+        {
+            _configuration = config ??
+                             throw new ArgumentNullException(nameof(config));
+            _mapper = mapper ??
+                      throw new ArgumentNullException(nameof(mapper));
+
+            repository = new AASRepositoryFactory(adtInteractions,mapper).CreateAASRepositoryForADT(config["ADT_SERVICE_URL"]) ??
+                         throw new ArgumentNullException(); ;
+        }
+
+
+        /// <summary>
+        /// Returns all Asset Administration Shells
+        /// </summary>
+        /// <param name="assetIds">The key-value-pair of an Asset identifier</param>
+        /// <param name="idShort">The Asset Administration Shell’s IdShort</param>
+        /// <response code="200">Requested Asset Administration Shells</response>
+        [HttpGet]
+        [ValidateModelState]
+        [SwaggerOperation("GetAllAssetAdministrationShells")]
+        [SwaggerResponse(statusCode: 200, type: typeof(List<AssetAdministrationShell>), description: "Requested Asset Administration Shells")]
+        public async Task<ActionResult<List<AssetAdministrationShell>>> GetAllAssetAdministrationShells([FromQuery(Name = "assetIds")] List<IdentifierKeyValuePair> assetIds, [FromQuery] string idShort)
```

Comment:
No exception handling? No standard error code handling according to the Standard?

Reply:
Not implemented yet